lib/plugins/aws/package/compile/events/apiGateway/lib/cors.js

```diff
@@ -98,9 +109,21 @@ module.exports = {
         StatusCode: '200',
         ResponseParameters: responseParameters,
         ResponseTemplates: {
-          'application/json': '',
+          'application/json':
+            Array.isArray(origins) ? this.generateCorsResponseTemplate(origins) : '',
         },
       },
     ];
   },
+
+  generateCorsResponseTemplate(origins) {
+    return (
```

Comment:
Does this only affect preflight requests or all requests? If the later, does it remove the need for users to specify that header manually in their handler?

Reply:
Only preflight. API Gateway does not support transformation of responses sent through lambda